leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -84,6 +87,10 @@ private AwaitingLeadershipProxy(Supplier<?> delegateSupplier,
         this.delegateRef = new AtomicReference<Object>();
         this.interfaceClass = interfaceClass;
         this.isClosed = false;
+
+        String logTag = "Proxy #" + numProxies.getAndIncrement() + " - " + interfaceClass.getSimpleName();
+        log = new TaggedLogger(AwaitingLeadershipProxy.class, logTag);
+        leaderLog = new TaggedLogger("leadership", logTag);
```

Comment:
Since the aim for this is to get it to a deployment quickly and gain signal, I think we should bias towards doing the dumb/quick thing, and then we can also get signal on whether the "clever/complex" approach is worth investing in.